### DISCUSSION TITLE
Skia - Changing BorderBrush of user control

### DISCUSSION THREAD

**Discussion by mcNets** [unoplatform/uno#9863] [Q&A]
Lets say I have a user control that mimics a LED with two possible states On / Off.

```
<Border x:Name="ST_border"
        BorderThickness="1"
        BorderBrush="Gray">

    <TextBlock x:Name="ST_text"
                HorizontalAlignment="Center"
                Foreground="White"
                Style="{StaticResource SysmacTextBlockH}" />
</Border>
```

So the control have a bool DP called IsON and 4 DP for each state:

- ControlTextON / ControlTextOFF
- BorderColorON / BorderColorOFF
- BackgraundColorON / BackgroundColorOFF
- TextColorON / TextColorOFF

When IsON changes, I simply change the value of these properties.

```
public bool IsON
{
    get { return (bool)GetValue(IsONProperty); }
    set
    {
        SetValue(IsONProperty, value);
        if (value)
        {
            this.ST_border.BorderBrush = BorderColorON;
            this.ST_border.Background = BackgroundColorON;
            this.ST_text.Foreground = TextColorON;
            this.ST_text.Text = ControlTextON;
        }
        else
        {
            this.ST_border.BorderBrush = BorderColorOFF;
            this.ST_border.Background = BackgroundColorOFF;
            this.ST_text.Foreground = TextColorOFF;
            this.ST_text.Text = ControlTextOFF;
        }
    }
}
```

That works fine in a Windows project, but when I run the Skia project I get this error:

>fail: Uno.UI.DataBinding.BindingPath+BindingItem[0]
      Failed to set the source value for [BorderBrush]
      System.InvalidOperationException: The color ControlStrokeColorOnAccentSecondaryBrush is unknown
         at Microsoft.UI.Colors.Parse(String colorCode) in C:\a\1\s\src\Uno.UI\UI\Colors.cs:line 62
         at Uno.UI.DataBinding.BindingPropertyHelper.FastStringToBrushConvert(Type outputType, String input, Object& output) in C:\a\1\s\src\Uno.UI\DataBinding\BindingPropertyHelper.FastConvert.cs:line 596
         at Uno.UI.DataBinding.BindingPropertyHelper.FastStringConvert(Type outputType, String input, Object& output) in C:\a\1\s\src\Uno.UI\DataBinding\BindingPropertyHelper.FastConvert.cs:line 206
         at Uno.UI.DataBinding.BindingPropertyHelper.FastConvert(Type outputType, Object input, Object& output) in C:\a\1\s\src\Uno.UI\DataBinding\BindingPropertyHelper.FastConvert.cs:line 56
         at Uno.UI.DataBinding.BindingPropertyHelper.Convert(Func`1 propertyType, Object value) in C:\a\1\s\src\Uno.UI\DataBinding\BindingPropertyHelper.cs:line 1075
         at Uno.UI.DataBinding.BindingPropertyHelper.<>c__DisplayClass40_9.<InternalGetValueSetter>b__13(Object instance, Object value) in C:\a\1\s\src\Uno.UI\DataBinding\BindingPropertyHelper.cs:line 858
         at Uno.UI.DataBinding.BindingPath.BindingItem.SetSourceValue(ValueSetterHandler setter, Object value) in C:\a\1\s\src\Uno.UI\DataBinding\BindingPath.cs:line 718

And none of the 4 properties change their value.

[ST_LED.zip](https://github.com/unoplatform/uno/files/9581705/ST_LED.zip)

**Comment by jeromelaban** [comment]
Thanks for the report! Make sure to include a full working repro project, we're not able to reproduce the issue with the extracts you provided.

**Reply by mcNets** [reply]
Yes, adding the minimum project to reproduce the error. Let me suggest to run first the Windows project.

Thanks.

[TestSkiaProperties.zip](https://github.com/unoplatform/uno/files/9584396/TestSkiaProperties.zip)

**Reply by jeromelaban** [reply]
Thanks for the repro! It was indeed a bug: https://github.com/unoplatform/uno/pull/9870

**Comment by mcNets** [comment]
I've tried using VisualStates and it fails too. Same error.

```
<VisualStateManager.VisualStateGroups>
    <VisualStateGroup x:Name="OnOff">
        <VisualState x:Name="On">
            <VisualState.StateTriggers>
                <StateTrigger IsActive="{x:Bind IsON, Mode=OneWay}" />
            </VisualState.StateTriggers>
            <VisualState.Setters>
                <Setter Target="ST_text.Foreground"
                        Value="{x:Bind TextColorON, Mode=OneWay}" />
                <Setter Target="ST_text.Text"
                        Value="{x:Bind ControlTextON, Mode=OneWay}" />
                <Setter Target="ST_border.Background"
                        Value="{x:Bind BackgroundColorON, Mode=OneWay}" />
                <Setter Target="ST_border.BorderBrush"
                        Value="{x:Bind BorderColorON, Mode=OneWay}" />
            </VisualState.Setters>
        </VisualState>
        <VisualState x:Name="Off">
            <VisualState.StateTriggers>
                <StateTrigger IsActive="{x:Bind IsOFF, Mode=OneWay}" />
            </VisualState.StateTriggers>
            <VisualState.Setters>
                <Setter Target="ST_text.Foreground"
                        Value="{x:Bind TextColorOFF, Mode=OneWay}" />
                <Setter Target="ST_text.Text"
                        Value="{x:Bind ControlTextOFF, Mode=OneWay}" />
                <Setter Target="ST_border.Background"
                        Value="{x:Bind BackgroundColorOFF, Mode=OneWay}" />
                <Setter Target="ST_border.BorderBrush"
                        Value="{x:Bind BorderColorOFF, Mode=OneWay}" />
            </VisualState.Setters>
        </VisualState>
    </VisualStateGroup>
</VisualStateManager.VisualStateGroups>
```
